deploy/operator/gather.sh

```diff
@@ -121,6 +151,7 @@ function gather_all() {
   gather_agent_data
   gather_clusterdeployment_data
   gather_imageset_data
+  gather_capi
```

Comment:
Consider calling gather_capi only when GATHER_CAPI_LOGS env is set (this is how it works in assisted-baremetal-gather)